turbine/src/retransmit_stage.rs

```diff
@@ -385,15 +402,27 @@ fn retransmit_shred(
                 .filter_map(|&addr| quic_endpoint_sender.try_send((addr, shred.clone())).ok())
                 .count()
         }
-        Protocol::UDP => match multi_target_send(socket, shred, &addrs) {
-            Ok(()) => addrs.len(),
-            Err(SendPktsError::IoError(ioerr, num_failed)) => {
-                error!(
+        Protocol::UDP => match socket {
+            RetransmitSocket::Xdp(sender) => {
+                let mut sent = num_addrs;
+                if num_addrs > 0 {
+                    if let Err(e) = sender.try_send(key.index(), addrs.to_vec(), shred) {
+                        log::warn!("xdp channel full: {e:?}");
```

Comment:
Should we get this patched into metrics ? if this starts happening, we are likely to see millions of warn! all at once, which will overwhelm logging.

Reply:
yes but then panic! might be more appropriate, as it will crash immediately with clear error rather than eating all of disk space and then crashing.

Reply:
Oh yeah? Let me offer this POV:

this should never happen, but considering the bursty behavior, it _might_ super occasionally happen? In that case, spamming a few warnings might be more appropriate than panicing say if it happens only once a month?

I don't care. I can panic if you want, but do you agree that this is exactly the definition of bikeshedding, considering that we know the real issue is the bursty nature of turbine, which I'd have time to fix if we didn't bikeshed here?

Reply:
No bikes involved. Your claim, as I understand it, is that this should never happen. This implies panic is appropriate. If you think this can occasionally happen, and the condition is recoverable then panic is not appropriate. But in this case we do not need per-shred warns in the logs, they are not helpful, we would much rather have metrics that count these.

Reply:
done

Reply:
Thank you. I do still wonder what would happen if we log::error! for every shred...

Reply:
we'll never know we log a warning 😏